### PR TITLE
add AsyncStorage wrapper from GoodDApp

### PR DESCRIPTION
**Pull request by johnsmith-gooddollar**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/sdk-v2/package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "@usedapp/core": "^1.0.9",
     "eventemitter3": "^4.0.7",
     "react-native": "^0.70.2",
+    "react-native-restart": "^0.0.24",
```
Comment (sirpy) [inline packages/sdk-v2/package.json:65]: why do we need restart?

Comment (johnsmith-gooddollar) [reply packages/sdk-v2/package.json:65]: i replaced location.reload with cross-platform implementation from GoodDApp
it uses restart on native

Comment (johnsmith-gooddollar) [reply packages/sdk-v2/package.json:65]: we could move it to the dev or peer deps

File: packages/sdk-v2/package.json [hunk 2]
```diff
@@ -26,7 +26,7 @@
     "@babel/preset-env": "^7.16.4",
     "@babel/preset-react": "^7.16.0",
     "@babel/preset-typescript": "^7.16.0",
-    "@react-native-async-storage/async-storage": "^1.17.10",
+    "@react-native-async-storage/async-storage": "1.17.10",
```
Comment (L03TJ3) [inline packages/sdk-v2/package.json:29]: 17.11 has been released 3 days ago which compiles with >ES2020 and for us to work with it on UI we have to upgrade to react-scripts@5
using 17.10 for now is quicker I think

**Review by L03TJ3 — COMMENTED** [review]
also, rebased to be merged to the 242-defaultcelo as this PR should be in sync with the 
UI pr: https://github.com/GoodDollar/GoodProtocolUI/pull/246

**Pull request merged**